rrtmgp/mo_gas_optics_rrtmgp.F90

```diff
@@ -1236,6 +1249,58 @@ function get_col_dry(vmr_h2o, plev, tlay, latitude) result(col_dry)
   end function get_col_dry
   !--------------------------------------------------------------------------------------------------------------------
   !
+  !
+  !
+  function compute_optimal_angles(this, optical_props, &
+                                         optimal_angles) result(err_msg)
+
+    ! input
+    class(ty_gas_optics_rrtmgp), &
+                                      intent(in   ) :: this
+    class(ty_optical_props_arry),     intent(in   ) :: optical_props  ! Gas volume mixing ratios
+
+    !output
+    character(len=128)   :: err_msg
+    real(wp), dimension(:,:), intent(inout)         :: optimal_angles
+
+    ! Local variables
+    !
+    integer  :: ncol, ngpt, nbnd
+    integer  :: col, gpt, bnd
+    real(wp) :: trans_total
+
+    ncol = optical_props%get_ncol()
+    ngpt = optical_props%get_ngpt()
+    nbnd = optical_props%get_nband()
+
+    err_msg=""
+
+    ! Check match in dimensions between tau and optimal_angle_fit
+    if (nbnd .ne. size(this%optimal_angle_fit,2)) err_msg = &
```

Comment:
Better to check that the optical properties and gas optics class have the same spectral discretization using `gpoints_are_equal()` from `ty_optical_props`

Reply:
Implemented this.